cpp/src/arrow/compute/kernels/scalar_string_internal.cc

```diff
@@ -0,0 +1,57 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <sstream>
+
+#include "arrow/compute/kernels/scalar_string_internal.h"
+
+namespace arrow {
+namespace compute {
+namespace internal {
+
+FunctionDoc StringPredicateDoc(std::string summary, std::string description) {
```

Comment:
This new file looks not necessary. Move these two functions to scalar_string_ascii.cc? Or to the header (static)?

Reply:
Those two functions are used by both scalar_string_ascii.cc and scalar_string_utf8.cc equally, so putting them in the header seems more suitable to me. I tried that initially by means of putting all the code in the header in an anonymous namespace, mimicking how the code used to work, but the linter and/or clang-tidy wouldn't have it. I'll give static a try.

Reply:
IMO, `scalar_string_internal.cc` should contain the definition of all the methods/functions in `scalar_string_internal.h`, instead of having them in embedded in `*.h` files. It makes it easier to understand the API of all those classes, but this is not followed systematically in Arrow and, if we want to do this, then it can be done in a follow-up PR.